### PR TITLE
[RCTDevMenu] Store live reload config outside of RCTDevMenu

### PR DESCRIPTION
**Pull request by brentvatne**
Config was being nuked each time a reload occurred because RCTDevMenu is dealloc'd and reinitialized, see #1067 

cc @tadeuzagallo @nicklockwood

### CODE REVIEW

File: React/Base/RCTDevMenu.h [hunk 1]
```diff
@@ -47,6 +47,24 @@
...
+@property (nonatomic, assign) NSTimeInterval liveReloadPeriod;
+@property (nonatomic, assign) BOOL liveReloadEnabled;
+
++ (RCTDevMenuConfiguration *)sharedInstance;
+
++ (NSTimeInterval)getLiveReloadPeriod;
++ (BOOL)getLiveReloadEnabled;
+
++ (void)setLiveReloadPeriod:(NSTimeInterval)liveReloadPeriod;
++ (void)setLiveReloadEnabled:(BOOL)liveReloadEnabled;
```
Comment (ide) [inline React/Base/RCTDevMenu.h]: getters and setters are implicitly declared by obj-c

File: React/Base/RCTDevMenu.h [hunk 2]
```diff
@@ -47,6 +47,24 @@
 
 @end
 
+@interface RCTDevMenuConfiguration : NSObject {
+  NSTimeInterval liveReloadPeriod;
+  BOOL liveReloadEnabled;
+}
```
Comment (ide) [inline React/Base/RCTDevMenu.h]: Can remove this, the property declarations below are sufficient

File: React/Base/RCTDevMenu.m [hunk 3]
```diff
@@ -36,6 +36,47 @@ - (void)RCT_motionEnded:(UIEventSubtype)motion withEvent:(UIEvent *)event
 
 @end
 
+@implementation RCTDevMenuConfiguration
+
+@synthesize liveReloadPeriod;
+@synthesize liveReloadEnabled;
```
Comment (ide) [inline React/Base/RCTDevMenu.m]: Can remove these, Obj-C autosynthesizes properties now

File: React/Base/RCTDevMenu.m [hunk 4]
```diff
@@ -36,6 +36,47 @@ - (void)RCT_motionEnded:(UIEventSubtype)motion withEvent:(UIEvent *)event
...
+
+static RCTDevMenuConfiguration *sharedObject;
++ (RCTDevMenuConfiguration *)sharedInstance
+{
+  if (sharedObject == nil) {
+    sharedObject = [[super allocWithZone:NULL] init];
```
Comment (ide) [inline React/Base/RCTDevMenu.m]: just alloc instead of allocWithZone is generally preferred

**Comment by brentvatne** [comment]
Thanks @ide - fixed those!

File: React/Base/RCTDevMenu.h (hunk 1 above)
Comment (brentvatne) [inline React/Base/RCTDevMenu.h]: Yeah I'm not sure how great of an idea that is but it seemed like the simplest way to store config independent of the `RCTDevMenu` instance. Diff above looks out of date still, I removed the redundant property declarations.

**Comment by brentvatne** [comment]
@tadeuzagallo - this would be a good fix to get into the next release if possible

**Comment by tadeuzagallo** [comment]
@nicklockwood has already landed internally a slightly different version of it, using `NSUserDefaults`, it should be out on the next sync.

**Comment by brentvatne** [comment]
@tadeuzagallo - excellent, will close this then

**Pull request closed without merging**